src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -4402,23 +4420,26 @@ void Parser::parse_and_visit_with(Parse_Visitor_Base &v) {
 
 void Parser::parse_and_visit_if(Parse_Visitor_Base &v) {
   QLJS_ASSERT(this->peek().type == Token_Type::kw_if);
-  Source_Code_Span if_token_span = this->peek().span();
+  Token if_token = this->peek();
   this->skip();
 
+  bool previous_in_if_statement = this->in_if_statement_;
+  this->in_if_statement_ = true;
```

Comment:
Must fix: Use `Bool_Guard`. (It's a bit of boilerplate, but the safety is worth it.) See `Switch_Guard` for example usage.